lib/OpenQA/Client/Archive.pm

```diff
@@ -0,0 +1,193 @@
+package OpenQA::Client::Archive;
+use strict;
+use warnings;
+use Data::Dump;
+
+use JSON;
+use Mojo::URL;
+use Mojo::File 'path';
+
+our $client;
+our %options;
+our $job;
+our $path;
+
+sub run {
+    #We're only taking one argument
+    ($client, %options) = @_;
+    my $method = 'get';
+
+    my $url = $options{url}->clone;
+    $url->path->merge('details');
+    my $req = $client->$method($url);
+    my $res = $req->res;
+
+    if ($res->code eq 200) {
+        $job = $res->json->{job};
+        #We have a job now, make a directory in $pwd by default
+        $path = path($options{archive} || $job->{name})->make_path;
+        chdir($path);
+        #we can't backup repos, so don't even try
+        delete($job->{assets}->{repos});
+        download_assets() unless $options{'skip-download'};
```

Comment:
I think I would make assets opt-in rather than opt-out - but with a size limit. So we by default download the small 'other' assets but skip ISOs and you can skip that in means of increasing the --asset-size-limit. I can't think of any use case keeping an archive of the HDDs.